docs/3.0-migration-guide.d/default-curves.md

```diff
@@ -0,0 +1,16 @@
+Weak curves are now disabled by default for X.509 and TLS
+---------------------------------------------------------
+
+The default X.509 verification profile (`mbedtls_x509_crt_profile_default`) and the default curve and hash selection have changed. X.509 and TLS now allow the same algorithms by default (except that the X.509 profile only lists curves that support signature verification).
+
+Hashes and curves weaker than 255 bits are no longer accepted by default. The following algorithms have been removed: SHA-1 (formerly only accepted for key exchanges but not for certificate signatures), SHA-224, secp192r1, secp224r1, secp192k1, secp224k1 (weaker hashes were already not accepted).
+
+The compile-time option `MBEDTLS_TLS_DEFAULT_ALLOW_SHA1_IN_KEY_EXCHANGE` is no longer available.
+
+If you still need to accept certificates signed with algorithms that have been removed from the default profile, call `mbedtls_x509_crt_verify_with_profile` instead of `mbedtls_x509_crt_verify` and pass a profile that allows the curves you want. For example, to allow SHA-224:
+```
+mbedtls_x509_crt_profile my_profile = mbedtls_x509_crt_profile_default;
+my_profile.allowed_mds |= MBEDTLS_X509_ID_FLAG( MBEDTLS_MD_SHA224 );
```

Comment:
Oh, that's pretty good. Perhaps the entry for `DEFAULT_ALLOW_SHA1_IN_CERTIFICATES` should be updated to suggest the same pattern. Or perhaps it would even make sense to merge both entries?

Reply:
In 2021, I don't want to explain to people how to enable SHA-1 in certificates. I deliberately only suggest not doing this anymore.

Reply:
Sorry, I wasn't talking about the entry you're adding, but about [this pre-existing entry](https://github.com/ARMmbed/mbedtls/blob/development/docs/3.0-migration-guide.md#remove-the-option-to-allow-sha-1-by-default-in-certificates). I think there are two possible changes we might want to make in that other entry:

- either just say "set a custom profile" without an example, because as you say, we don't really want to help people doing that in 2021;
- or, if we want to keep an example of how to set a custom profile, I think the method you're showing it is superior to the one in the pre-existing SHA-1 entry.

Also, I was wondering if it would make sense to completely remove the SHA-1 entry and merge it into this one. I'm not saying we should definitely do it, but I wanted to mention that option.